### PR TITLE
General refactors and improvements

### PR DESCRIPTION
**Pull request by juanimedone**
The main changes of this PR include:

- Adding more instructions tests.
- Including VM's installation (in `README.md`) and updating usage.
- Refactoring memory reading from `stdin` and `read_image_file`.
- Turning 'memory' and 'registers' from `Vec`s to fixed arrays.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,15 +11,20 @@ This project is an implementation of a simple LC-3 (Little Computer 3) virtual m
...
+    ```bash
+    cargo install --path .
+    ```
+
+3. Run the VM.
 
+    You can run the VM with one or more LC-3 object files as arguments. 
     ```bash
-    cargo run object-file1 object-file2 ...
+    lc3-vm object-file1 object-file2 ...
```
Comment (belfortep) [inline README.md:23]: I think we could add an explanation of how to pass from one program to another, just tried it and after one program ends it just returns to the terminal

Comment (belfortep) [reply README.md:23]: I think the problem is when we start the program, we call the `read_image_file` function with all the `paths` from the `main`, and it loads everything in the same memory space, using only the last object-file

**Review by lazcanoluca — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (lazcanoluca) [reply README.md:23]: I think as the programmer you are supposed to account for that when writing the assembly. Specifically, with the assembler directives (indicating where the program begins in memory).

**Review by lazcanoluca — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (belfortep) [reply README.md:23]: Oh, didn't think about that case, then there is no problem 🤝

**Review by belfortep — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (juanimedone) [reply README.md:23]: > I think as the programmer you are supposed to account for that when writing the assembly. Specifically, with the assembler directives (indicating where the program begins in memory).

Yes, that's the idea.

**Pull request merged**